src/sentry/deletions/defaults/group.py

```diff
@@ -204,19 +206,45 @@ def delete_events_from_nodestore(self, events: Sequence[Event]) -> None:
     def delete_events_from_snuba(self) -> None:
         requests = []
         for project_id, group_ids in self.project_groups.items():
```

Comment:
If we have > 2m groups here I'm kind of surprised we're not OOMing. Are you sure that what it's complaining about is based on the number of groups? Could it be the number of events/occurrences inside the groups instead?

Reply:
The number of events. This is why I'm slicing based on `times_seen`.

Reply:
Ahh, I missed that, makes sense.